draft-jmiller-json-proof-algorithms.md

```diff
@@ -73,112 +79,127 @@ Due to the nature of ZKPs, this specification also documents the subtle but impo
+In order to support the protection of a presentation by a holder to a verifier, the holder MUST use a Replay Key during the issuance and the presentation of every Single Use JWP.  This Replay Key MUST be generated and used for only one JWP.
 
-The signer must choose only an asymmetric JWS algorithm so that each signature is non-deterministic.  This ensures that no other party can brute-force any non-disclosed payloads based only on their individual signatures.
+The issuer MUST verify that the holder has possession of this key.  The holder-issuer communication to exchange this information is out of scope of this specification, but can be easily accomplished by the holder using this key to generate a JWS that signs a value the issuer can verify as unique.
+
+### Issuer Setup
+
+To create a Single Use JWP the issuer first generates a unique Ephemeral Key using the selected JWS algorithm.  This key-pair will be used to sign each of the payloads of a single JWP and then discarded. 
 
 ### Using JWS
 
-JSON Web Signatures are used to create the signature values used by the SU algorithm.  This allows an implementation to use an existing JWS library directly for all necessary cryptographic operations without requiring any additional primitives.
+JSON Web Signatures are used to create all of the signature values used by the SU algorithm.  This allows an implementation to use an existing JWS library directly for all necessary cryptographic operations without requiring any additional primitives.
+
+Each individual JWS uses a fixed protected header containing only the minimum required `alg` value.  Since this JWS protected header itself is the same for every JWS, it SHOULD be a static value in the form of `{"alg":"***"}` where `***` is the JWA asymmetric signing key algorithm identifier being used.  This value is re-created by a verifier using the correct JWA algorithm value included in the SU algorithm identifier.
 
-Each individual JWS uses a fixed  protected header containing only the minimum required `alg` value.  Since this JWS protected header itself is the same for every JWS, it SHOULD be a static value in the form of `{"alg":"***"}` where `***` is the JWS asymmetric key algorithm being used.  This value is re-created by a Verifier using the correct algorithm value.
+If an implementation uses an alternative JWS protected header than this fixed value, a base64url encoded serialized form of the alternate fixed header MUST be included using the `jws_header` claim in the issuer protected header.
 
-If an implementation uses an alternative JWS protected header than the fixed value then a base64url encoded serialized form of that fixed header MUST be included as the `proof_header` value in the JWP protected header.
+### Issuer Protected Header
 
-### Protected Header
+The JWK of the issuer's Ephemeral Key MUST be included in the issuer protected header with the property name of `proof_jwk` and contain only the REQUIRED values to represent the public key.
```

Comment:
hmm, i think `jwk` is already  a registered term for header use... I wonder if we need the proof prefix.